src/test/java/org/springframework/integration/aws/outbound/S3MessageHandlerTests.java

```diff
@@ -255,19 +260,25 @@ public void testDownloadDirectory() throws IOException {
 
 		this.s3SendChannel.send(message);
 
-		File[] fileArray = directoryForDownload.listFiles();
+		// get the "root" directory
+		File[] directoryArray = directoryForDownload.listFiles();
+		assertThat(directoryArray).isNotNull();
+		assertThat(directoryArray.length).isEqualTo(1);
+
+		// get the files we downloaded
+		File[] fileArray = directoryArray[0].listFiles();
```

Comment:
I added 
```
File subDirectory = directoryArray[0];
assertThat(subDirectory.getName()).isEqualTo("subdir");
```
to be sure that we restore the remote hierarchy properly.
Also added your name to this class.

Merging...